src/commands/typescript/update.ts

```diff
@@ -67,37 +106,78 @@ export default class Update extends SfdxCommand {
     await fs.writeJson(tsConfigPath, tsConfig);
   }
 
-  private async updateTsVersion(): Promise<void> {
-    const newVersion = this.determineNextTsVersion();
-    const pkg = await Package.create(path.resolve('.'));
-
+  private async updatePackage(packagePath: string, npmPackage: NpmPackage): Promise<void> {
+    const newVersion = this.determineNextTsVersion(npmPackage);
+    const pkg = await Package.create(path.resolve(packagePath));
     if (pkg.packageJson.devDependencies['typescript']) {
       this.ux.log(`Updating typescript version to ${newVersion}`);
       pkg.packageJson.devDependencies['typescript'] = newVersion;
       pkg.packageJson.devDependencies['@typescript-eslint/eslint-plugin'] = 'latest';
       pkg.packageJson.devDependencies['@typescript-eslint/parser'] = 'latest';
     }
-
     // If the prepare script runs sf-install, the install will fail because the typescript version
     // won't match the expected version. So in that case, we delete the prepare script so that we
     // get a successful install
     if (pkg.packageJson.scripts['prepare'] === 'sf-install') {
       delete pkg.packageJson.scripts['prepare'];
     }
+    return pkg.writePackageJson();
+  }
 
-    pkg.writePackageJson();
+  private async updateTsVersion(): Promise<void> {
+    if (this.isLernaRepo()) {
+      const workingDir = pwd().stdout;
+      const packagePaths = await this.getPackagePaths();
+
+      for (let i: number; i < packagePaths.length; i++) {
+        const packagePath = packagePaths[i];
+        this.ux.log('packagePath:', packagePath);
+        cd(path.resolve(packagePath));
+        const runeer = exec('npm view typescript --json', { silent: true });
+        cd(workingDir);
+        if (runeer.code === 0) {
+          const npmPackage = JSON.parse(runeer.stdout) as NpmPackage;
+          await this.updatePackage(packagePath, npmPackage);
+        } else {
+          throw new SfdxError('Could not find typescript on the npm registry', 'TypescriptNotFound');
+        }
+      }
+      return;
+    }
+
+    this.typescriptPkgs.map(async (typescriptPkg) => {
+      this.ux.log(typescriptPkg.version);
+      await this.updatePackage(path.resolve('.'), typescriptPkg);
+    });
   }
 
-  private determineNextTsVersion(): string {
+  private determineNextTsVersion(typescriptPkg: NpmPackage): string {
     return this.flags.version === 'latest' || !this.flags.version
-      ? getString(this.typescriptPkg, 'dist-tags.latest')
+      ? getString(typescriptPkg, 'dist-tags.latest')
       : (this.flags.version as string);
   }
 
-  private retrieveTsPackage(): NpmPackage {
-    const result = exec('npm view typescript --json', { silent: true });
-    if (result.code === 0) {
-      return JSON.parse(result.stdout) as NpmPackage;
+  private async retrieveTsPackages(): Promise<NpmPackage[]> {
```

Comment:
This is not right. There is trying to return the package.json for the [typescript](https://www.npmjs.com/package/typescript) package. There is only one. There is no need to get it for every lerna package dir, because it will be exactly the same. i.e. the `npm view typescript --json` will return the same thing on every call...